kDriveCore/Data/Api/Endpoint.swift

```diff
@@ -135,29 +135,18 @@ public struct ProxyFile: AbstractFile, Sendable {
 
     /// Resolve an abstract file within a `DriveFileManager`.
     func resolve(within driveFileManager: DriveFileManager) throws -> File {
-        let file = driveFileManager.fetchObject(ofType: File.self) { realm in
-            fetch(using: realm)
-        }
-
-        guard let file, !file.isInvalidated else {
+        guard let file = driveFileManager.fetchObject(ofType: File.self, forPrimaryKey: uid),
+              !file.isInvalidated else {
             throw DriveError.errorWithUserInfo(.fileNotFound, info: [.fileId: ErrorUserInfo(intValue: id)])
         }
 
         return file
     }
 
-    /// Internal query an object from a realm
-    private func fetch(using realm: Realm) -> File? {
-        guard let file = realm.object(ofType: File.self, forPrimaryKey: uid), !file.isInvalidated else {
-            return nil
-        }
-
-        return file
-    }
-
     /// Resolve an abstract file within a `realm`. Throws if not found.
     func resolve(using realm: Realm) throws -> File {
-        guard let file = fetch(using: realm) else {
+        guard let file = realm.object(ofType: File.self, forPrimaryKey: uid),
+              !file.isInvalidated else {
```

Comment:
Looks like this code is implemented differently than on the initial branch. Waiting for initial branch merge then rebase then I will take a look again